### PR TITLE
code structure update

### LINKED ISSUE #52: docstring addition!

### PR DESCRIPTION
**Pull request by rajveer43**
close #52 

@alxmrs please review this PR

### CODE REVIEW

File: docs/moisture_dataset.py [hunk 1]
```diff
@@ -10,12 +10,18 @@
...
 
 def attribute_fix(ds):
-    """Needed to fix a low-level bug in ecCodes.
-    
-    Sometimes, shortNames get overloaded in ecCodes's table. 
-    To eliminate ambiguity in their string matching, we
-    force ecCodes to make use of the paramId, which is a
-    consistent source-of-truth.
+    """
+    Fix attributes in an xarray Dataset for compatibility with ecCodes.
```
Comment (alxmrs) [inline docs/moisture_dataset.py:14]: Please make this continue from the first line. e.g. 
"""Fix attributes...

File: docs/moisture_dataset.py [hunk 2]
```diff
@@ -27,12 +33,38 @@ def attribute_fix(ds):
 
 
 def compute_gg_ll_dataset(ml_surface, ml_wind, ml_moisture, datestring):
-   """Merges/processes AR ML dataset to a CO ML dataset.
-   
-   This function takes a datestamp and the three AR model-level ERA5 
-   datasets to produce a unified model-level dataset on a Gaussian Grid
-   and a regular lat-lon grid for a single hour.
    """
+    Merges and processes AR model-level ERA5 datasets into CO ML Gaussian Grid and regular lat-lon grid datasets for a single hour.
```
Comment (alxmrs) [inline docs/moisture_dataset.py:37]: Same as above; please apply to all.

**Review by alxmrs — COMMENTED** [review]
Added a nit. Once fixed, I'll be happy to merge. 

Thanks for your contribution!

File: src/arco_era5/source_data.py [hunk 3]
```diff
@@ -184,7 +243,19 @@ def read_multilevel_vars(year,
...
+        root_path (str): Root directory where the NetCDF files are located.
+
+    Returns:
+        dict: A dictionary containing variable attributes.
+
+    This function retrieves attributes for all variables from NetCDF files.
+
+    Example:
+        >>> variable_attrs = get_var_attrs_dict()
+    """"""Returns attributes for all variables."""
```
Comment (dabhicusp) [inline src/arco_era5/source_data.py:258]: @rajveer43 can you modify this line?

File: docs/util.py [hunk 4]
```diff
@@ -130,7 +192,54 @@ def plot_shaded_with_barbs(plot_ds,
     fig.colorbar(im, cax=cax)
 
 
+
```
Comment (dabhicusp) [inline docs/util.py:195]: @rajveer43 can you remove this line as in standard practice only 2 empty lines between any function here I found 3 empty lines.

File: setup.py [hunk 5]
```diff
@@ -38,6 +38,7 @@
 # This class handles the pip install mechanism.
 class build(_build):  # pylint: disable=invalid-name
     """A build command class that will be invoked during package install.
+
```
Comment (dabhicusp) [inline setup.py:41]: Can you justify why this empty line here??

**Comment by dabhicusp** [comment]
Added my suggestions.
Is there any changes needed @alxmrs ? what you say?

BTW Thank you for the contribution! @rajveer43.

File: docs/moisture_dataset.py [hunk 6]
```diff
@@ -27,12 +32,37 @@ def attribute_fix(ds):
 
 
 def compute_gg_ll_dataset(ml_surface, ml_wind, ml_moisture, datestring):
```
Comment (alxmrs) [inline docs/moisture_dataset.py:34]: In a future PR, let's also add type annotations to these vars.

**Review by alxmrs — APPROVED** [review]
LGTM! I would love input from @wundersooner, who may be able to double check some if the `mositure_dataset.py` comments make sense.

**Review by dabhicusp — COMMENTED** [review]
LGTM!

**Review by dabhicusp — APPROVED** [review]

**Pull request merged**